### PR TITLE
Use Object.assign instead of $.extend

### PR DESCRIPTION
**Pull request by Clarity-89**

### CODE REVIEW

File: docs/rest_client/models.rst [hunk 1]
```diff
@@ -20,10 +20,12 @@ A model needs just enough information to be able to communicate with the API.
 
    import { Model } from 'ponyjs/models.js';
 
-   // provide the model name + declaration of fields/meta/managers
+   // Provide the model name + declaration of fields/meta/managers
+   // If `model_name` is omitted, lower case representation of the model's name is used
```
Comment (sergei-maertens) [inline docs/rest_client/models.rst:24]: maybe move this as a comment to the actual line where `model_name` is specified?

File: src/ponyjs/models/base.js [hunk 2]
```diff
@@ -1,16 +1,13 @@
 'use strict';
 
-import $ from 'jquery';
```
Comment (sergei-maertens) [inline src/ponyjs/models/base.js:3]: *parties* \o/

File: src/ponyjs/models/base.js [hunk 3]
```diff
@@ -56,14 +53,12 @@ class ModelState {
...
-  constructor(data={}) {
+  constructor(data) {
 
     // set up the instance state
     this._state = data;
 
-    for (let key of Object.keys(data)) {
-      this[key] = data[key]
-    }
+    Object.assign(this, data);
```
Comment (sergei-maertens) [inline src/ponyjs/models/base.js:61]: I wonder what happens now if `data` is `undefined`, do you know if there's a test for that case?

File: src/ponyjs/models/base.js (hunk 2 above)
Comment (Clarity-89) [reply src/ponyjs/models/base.js:3]: :3

File: docs/rest_client/models.rst (hunk 1 above)
Comment (Clarity-89) [reply docs/rest_client/models.rst:24]: I tried, but it seemed a bit too long and was wrapping on several lines, so it didn't look that good to me.

**Comment by sergei-maertens** [comment]
Ah yeah, leave it as is then :-)

File: src/ponyjs/models/base.js (hunk 3 above)
Comment (Clarity-89) [reply src/ponyjs/models/base.js:61]: Object.assign doesn't throw an error when source is undefined, so it all works, and yes there're 3 test cases where it is undefined so it's covered.

**Comment by sergei-maertens** [comment]
Good stuff

**Pull request merged**